File: javascript/gemini/p123.js
// 1. Check for exactly one argument
if (process.argv.length !== 3) {
    console.log("Error: Provide exactly one argument.");
    process.exit(1);
}

let arg = process.argv[2];

// 2. Check if it's a string
if (typeof arg !== 'string') {
    process.exit(1);
}

let charArray = arg.split('');

// 3. Trim trailing whitespace
for (let i = charArray.length - 1; i >= 0; i--) {
    if (/\s/.test(charArray[i])) {
        charArray.pop();
    } else {
        break;
    }
}

// 4. Trim newlines
for (let i = 0; i < charArray.length; i++) {
    if (charArray[i] === '\n' || charArray[i] === '\r') {
        charArray[i] = '';
    }
}

// 5. Join (JS handles the rest)
const result = charArray.join('');
console.log(`Result: '${result}'`);

File: python/gemini/p123.py
import sys

# 1. Check for exactly one argument
if len(sys.argv) != 2:
    print("Error: Provide exactly one argument.")
    sys.exit(1)

arg = sys.argv[1]

# 2. Check if argument is a string (sys.argv elements are strings by default)
if not isinstance(arg, str):
    sys.exit(1)

# Convert to list for "manual" manipulation
chars = list(arg)

# 3. Trim trailing whitespace
for i in range(len(chars) - 1, -1, -1):
    if chars[i].isspace():
        chars.pop()
    else:
        break

# 4. Trim newlines
for i in range(len(chars)):
    if chars[i] == '\n':
        chars[i] = '' # Removing newline

# 5. Resulting string (Python handles null termination internally)
result = "".join(chars)
print(f"Result: '{result}'")